modules/auxiliary/scanner/msmail/exchange_enumerator.go

```diff
@@ -0,0 +1,543 @@
+//usr/bin/env go run "$0" "$@"; exit "$?"
+
+/*
+ OWA (Outlook Webapp) is vulnerable to time-based user enumeration attacks.
+ This module leverages all known, and even some lesser-known services exposed by default
+ Exchange installations to enumerate users. It also targets Office 365 for error-based user enumeration.
+
+ Identify Command
+
+    Used for gathering information about a host that may be pointed towards an Exchange or o365 tied domain
+    Queries for specific DNS records related to Office 365 integration
+    Attempts to extract internal domain name for onprem instance of Exchange
+    Identifies services vulnerable to time-based user enumeration for onprem Exchange
+    Lists password-sprayable services exposed for onprem Exchange host
+
+ Userenum (o365) Command
+
+    Error-based user enumeration for Office 365 integrated email addresses
+
+ */
+package main
+
+import (
+	"crypto/tls"
+	b64 "encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"metasploit/module"
+	"net"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+	"strconv"
+)
+
+func main() {
+	metadata := &module.Metadata{
+		Name:        "msmailprobe",
+		Description: "Office 365 and Exchange Enumeration",
+		Authors:     []string{"poptart", "jlarose", "Vincent Yui", "grimhacker", "Nate Power", "Nick Powers", "clee-r7"},
+		Date:        "2018-11-6",
+		Type:        "single_scanner",
+		Privileged:  false,
+		References:  []module.Reference{},
+		Options: map[string]module.Option{
+			"Command":          {Type: "string", Description: "Either 'userenum' or 'identify'", Required: true, Default: "identify"},
```

Comment:
Metasploit's proper terminology for this is an 'action'. Please use that instead of 'command' so it works like other modules. We may need to extend the framework so that this shows up in the right place in the metadata.